src/Spd.Manager.Licence/BizPortalUserManager.cs

```diff
@@ -54,6 +55,38 @@ public async Task<BizPortalUserResponse> Handle(BizPortalUserCreateCommand reque
         return _mapper.Map<BizPortalUserResponse>(response);
     }
 
+    public async Task<BizPortalUserResponse> Handle(BizPortalUserUpdateCommand request, CancellationToken ct)
+    {
+        PortalUserListResp existingUsersResult = await _portalUserRepository.QueryAsync(
+            new PortalUserQry()
+            {
+                OrgId = request.BizPortalUserUpdateRequest.BizId,
+                ContactRoleCode = new List<ContactRoleCode> { ContactRoleCode.PrimaryBusinessManager, ContactRoleCode.BusinessManager },
+                PortalUserServiceCategory = PortalUserServiceCategoryEnum.Licensing
+            },
+            ct);
+
+        //check if email already exists for the user
+        if (existingUsersResult.Items.Any(u => u.UserEmail != null && u.UserEmail.Equals(request.BizPortalUserUpdateRequest.Email, StringComparison.InvariantCultureIgnoreCase)))
+        {
+            throw new DuplicateException(HttpStatusCode.BadRequest, $"User email {request.BizPortalUserUpdateRequest.Email} has been used by another user");
+        }
+
+        //check if user id in request exists in result list
+        var existingUser = existingUsersResult.Items.FirstOrDefault(u => u.Id == request.BizPortalUserUpdateRequest.Id);
+        if (existingUser == null)
+            throw new NotFoundException(HttpStatusCode.BadRequest, $"Cannot find the user");
+
+        BizResult biz = await _bizRepository.GetBizAsync(request.BizPortalUserUpdateRequest.BizId, ct);
+        int primaryUserNo = existingUsersResult.Items.Count(u => u.ContactRoleCode == ContactRoleCode.PrimaryBusinessManager);
```

Comment:
you count existing primary user no, not the number after user updated it.

Reply:
Fixed